habrakarma_

```diff
@@ -15,8 +15,8 @@ rating.label Rating"
 fi
 
 exec 33<>/dev/tcp/habrahabr.ru/80
-echo -e "GET /api/profile/$username/ HTTP/1.0\r\nHost: habrahabr.ru\r\n" >&33
-read -t 5 -d $'\0' REPLY <&33
+printf "%s\r\n" "GET /api/profile/$username/ HTTP/1.0" "Host: habrahabr.ru" "" >&33
+read -t 5 -d "\0" REPLY <&33
```

Comment:
А вот тут вы поломали :wink: 
```bash
$ echo -n $'\n' | hexdump -C
00000000  0a                                                |.|
00000001
$ echo -n "\n" | hexdump -C
00000000  5c 6e                                             |\n|
00000002
```
Смотрим `man bash` про `read`:
> -d delim
> The first character of delim is used to terminate the input line, rather than newline.
Мне в этой строке нужно просто запихнуть весь выхлоп в переменную, не разделяя его на строки. С вашим изменением read пытается делить на строки по символу '\'. По некоторой случайности этот символ в XMl не встречается и ваш патч работает, но не гуд....

А вообщем по-хорошему стоит сделать так, чтоб вообще не делить на строки без малоизвестной магии:
```bash
read -t 5 -d '' REPLY <&33
```